### ISSUE TITLE
Reflected XSS vulnerability (2MD)

### ISSUE THREAD

**Issue by miguelcobain** [nhn/tui.editor#386]
## Version

1.3.0

## Test Environment

Firefox 65. MacOS High Sierra

## Current Behavior

Right now, in any editor instance, users can "accidentally" XSS attack themselves. This is called [Reflected XSS](https://security.stackexchange.com/questions/65142/what-is-reflected-xss).

To reproduce it, you can go to any editor demo like this one: https://nhnent.github.io/tui.editor/api/latest/tutorial-example01-basic.html

Then type
```html
<? foo="><script>alert(1)</script>">
<! foo="><script>alert(1)</script>">
</ foo="><script>alert(1)</script>">
<svg onload="javascript:alert(document.cookie)" xmlns="#"></svg>
```

You should see an alert popping up with cookie contents (which are non existent on the demo site, but could have sensitive info on our web apps).

Only seems to work on Firefox. Chrome doesn't display the alert.

From my research, this seems to come from this line: https://github.com/nhnent/tui.editor/blob/master/src/js/convertor.js#L131

The unescaped content is added to the previously created div and that triggers user-entered code.

## Expected Behavior

The alert shouldn't be triggered, in this example.

**Comment by sohee-lee7** [comment]
@miguelcobain Thank you for report. tui.editor sterilizes html, but there seems to be a problem in Firefox. If you can afford it, I'd appreciate it if you could give PR on this issue.

**Comment by miguelcobain** [comment]
I'd love to, but I'm not sure what the `_removeBrToMarkPassAttributeInCode` is doing.

The error comes from adding unsanitized html to a detached element created with `const $wrapperDiv = $('<div />');`.

Maybe we could sanitize for doing that?

**Comment by sohee-lee7** [comment]
I think that Firefox evaluate the script in the html string when the html string is parsed in below case.
```js
const $wrapperDiv = $('<div />');
$wrapperDiv.html(html);
```
When type the text in markdown, the text of markdown is rendered by MarkdownIt. In this process, we insert `data-tomark-pass` in the `<br>` and then remove that in codeblock case. When removing `data-tomark-pass` in the codeblock, the string html is parsing to Dom (this is that `_removeBrToMarkPassAttributeInCode` function do and cause that display alert). This problem was resolved by another issue #403 .

But when change to Wysiwyg, the alert still display. `_removeBrToMarkPassAttributeInCode` function should be moved after sanitize that @miguelcobain said.  I will soon resolve that.

**Comment by miguelcobain** [comment]
Thank you! 👍
